File: bothsame.py
def count(s):
    dic = dict()
    for i in range(len(s)):
        if s[i] in dic:
            dic[s[i]] += 1
        else:
            dic[s[i]] = 1
    res = 0
    for i in dic.values():
        res += (i + 1) * i // 2
    return res


if __name__ == "__main__":
    print(count("aaa")) # 6
    print(count("abcd")) # 4
    print(count("ababca")) # 10

File: Java_template/BothSame.java
public class BothSame {
    public long count(String s) {
        // TODO
    }

    public static void main(String[] args) {
        BothSame b = new BothSame();
        System.out.println(b.count("aaa")); // 6
        System.out.println(b.count("abcd")); // 4
        System.out.println(b.count("ababca")); // 10
    }
}
